contrib/niti/niti.nit

```diff
@@ -0,0 +1,157 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# An interactive Nit interpreter
+module niti
+
+import nitc::interpreter
+import nitc::frontend
+import nitc::parser_util
+import readline
+
+redef class ToolContext
+	# Parse a full module given as a string
+	#
+	# Return a AModule or a AError
+	fun p_module(string: String): ANode
+	do
+		var source = new SourceFile.from_string("", string)
+		var lexer = new Lexer(source)
+		var parser = new Parser(lexer)
+		var tree = parser.parse
+
+		var eof = tree.n_eof
+		if eof isa AError then
+			return eof
+		end
+		return tree.n_base.as(not null)
+	end
+
+	# Parse the input of the user as a module
+	fun i_parse(prompt: String): nullable ANode
+	do
+		var oldtext = ""
+
+		loop
+			var s = readline(prompt)
+			if s == null then return null
+			if s == "" then continue
+
+			if s.chars.first == ':' then
```

Comment:
`:` is a unused placeholder, I plan to use a most nitish interface with the interactive interpreter.
A runtime library can provide methods like `help` or `quit` or native methods shorcuted by the interpreter to provide some meta power.